stdlib/public/core/SmallString.swift

```diff
@@ -127,25 +127,30 @@ extension _SmallUTF8String {
 #endif
   }
 
+  @inline(__always)
   @inlinable
+  @effects(readonly)
   public // @testable
   init?(_ codeUnits: UnsafeBufferPointer<UInt8>) {
 #if arch(i386) || arch(arm)
     return nil // Never form small strings on 32-bit
 #else
     let count = codeUnits.count
     guard count <= _SmallUTF8String.capacity else { return nil }
-    self.init()
-    self._withAllUnsafeMutableBytes { rawBufPtr in
-      let rawDst = rawBufPtr.baseAddress._unsafelyUnwrappedUnchecked
-      memcpy_(
-        dst: rawDst.assumingMemoryBound(to: UInt8.self),
-        src: codeUnits.baseAddress._unsafelyUnwrappedUnchecked,
-        count: count
-      )
+
+    let addr = codeUnits.baseAddress._unsafelyUnwrappedUnchecked
+    var high: UInt
+    let lowCount: Int
+    if count > 8 {
+      lowCount = 8
+      high = _bytesToUInt(addr + 8, count &- 8)
+    } else {
+      lowCount = count
+      high = 0
     }
-    _sanityCheck(self.count == 0, "overwrote count early?")
-    self.count = count
+    high |= (UInt(count) &<< (8*15))
```

Comment:
Can you instead put this in the setter for `count`, so that all other code paths can benefit?

Also, rather than have a naked `15`, you can use `capacity`.

Reply:
sure

Reply:
Actually, the count setter would have to mask out the old value, which is not needed here.
Also, I followed the count getter+setter pattern which also hard-code the '15'.